### ISSUE TITLE
[SSC] Invalid entries captured in snapshot

### ISSUE THREAD

**Issue by dostrelith678** [cardano-community/koios-artifacts#91]
**Describe the bug**
The stake snapshot values captured do not match the epoch_stake table.

**To Reproduce**
Steps to reproduce the behaviour:
```
cexplorer=# SELECT COUNT(*) FROM EPOCH_STAKE WHERE EPOCH_NO = 362;
  count
---------
 1215933
(1 row)

cexplorer=# SELECT COUNT(*) FROM grest.STAKE_SNAPSHOT_CACHE WHERE EPOCH_NO = 360;
  count
---------
 1215959
(1 row)

cexplorer=# SELECT SUM(amount) FROM EPOCH_STAKE WHERE EPOCH_NO = 362;
        sum
-------------------
 24724295922908154
(1 row)

cexplorer=# SELECT SUM(amount) FROM grest.STAKE_SNAPSHOT_CACHE WHERE EPOCH_NO = 360;
        sum
-------------------
 24724295975715173
(1 row)
```

We have some extra entries in the snapshot that should not be captured:
```
select sa.view, ph.view, ssc.* from grest.stake_snapshot_cache ssc inner join stake_address sa ON sa.id = ssc.addr_id INNER JOIN pool_hash ph on ph.id = ssc.pool_id where epoch_no = 360 and ssc.addr_id NOT IN (SELECT addr_id from epoch_stake where epoch_no = 362);
                            view                             |                           view                           | addr_id | pool_id |  amount  | epoch_no
-------------------------------------------------------------+----------------------------------------------------------+---------+---------+----------+----------
 stake1u8dxcszp4xnfafanzuduwusa9pjsy3x3hk3lurj6ck5z70ghlst7v | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |    4470 |     221 |        0 |      360
 stake1u8n2cakupn552njqx6nth5q6a8ssm7js2xr4g0mfaupxngqrh6mtk | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   99360 |     221 |        0 |      360
 stake1u84v56pv66er9lxkmv33glk2s5uuhxa0dc6tfgkgnxa8z7ga5euwj | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   22023 |     221 |        0 |      360
 stake1u9tx44879lxa9vknvc2e60jv0lnwrj49pf3qmjkgldpv74ce2urfm | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   87938 |     221 |        0 |      360
 stake1u9yvmkcx2ftyq2g4lu060tjp929kxwgdv38mdgww36qjvzsf5ryhe | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   99445 |     221 |        0 |      360
 stake1u94ly60w6wx50kl5rzcrf6t4pf4ltgpwgc6haeglvcdxwhsqnvy7w | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   99458 |     221 |        0 |      360
 stake1uysd603jll45ev3lp7e63l3kzlj6mq6grgy2w9s667pccxcy8szhk | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   99489 |     221 |        0 |      360
 stake1u9amhdfvddtcfzzmvf2qurg27f9f945vvhgrqaa8vyr2z6cutpu6c | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |   99512 |     221 |        0 |      360
 stake1u9pgavfu9p3dkpmqes7kr6gzmlwq0r0egvhcm4a04808nzg9x7tfq | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  100649 |     221 |        0 |      360
 stake1ux073fa7xs49jljgd8pceauymhgneqrc7vgsenhnqg93emgdlnjq4 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  123512 |     221 |        0 |      360
 stake1u8v4uwrlr8aaeujlqp96k43hmzpcanawehga4p2fzukxjaqz884p7 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  113216 |     221 | 20432252 |      360
 stake1uycf7qf4mlz3hhhrqgs65yeykn0nf3qq294spa432gc6trce2vvx7 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  131745 |     221 |    13627 |      360
 stake1uylsedsn3ul90xyc23nxyee0k3g6lfyx50yds4m9xza9trqae8ae0 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  129062 |     221 |   439724 |      360
 stake1uyq2qk7le8l9vwsd8l9s5khuhqrcdmta8mxvfjs35lcne5q4tcl59 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  131889 |     221 |        0 |      360
 stake1u9v8gj329v47t9ew4w64g4m8vcps0nj7q6hpl6jl8wxu5jqgrrm4y | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  144510 |     221 |   230492 |      360
 stake1ux05gxdgq4adx5v325750f9rddcd7czzcsxzs864rl0623q7h3yez | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  181019 |     221 |   995534 |      360
 stake1u82neuuzrswsjm50kq5gxjkkz57lrhpthfvmx0pk737ms9s2f96w9 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  177232 |     221 | 16312283 |      360
 stake1uxhsakr3nsnz4578cn88zkayk0gvjdrqq4qc9z5wvufvllgrmslvq | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  152656 |     221 |     6845 |      360
 stake1u8gl0c7n0c9uj4c4nlugxr5thn3r0mf5n85u53zezs269gszh5gtc | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  185599 |     221 |        0 |      360
 stake1uydj64zzepvap2k7ks8ut0y99dy08umap5aptfxjycs3raq582ng2 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  208121 |     221 |        0 |      360
 stake1uyns7pa8p49rds0gan03kvg2qrgxdhv0fgwg8gwqge5vp0qxeus0m | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  208155 |     221 |        0 |      360
 stake1u928ahgs6hxymkw2neys9q0d64uhu652gll49d80xqycjqczgz5wh | pool15gdh4w26ax66aa2j5ehwcvm46as3jx0v7mhu678y7m5q5s80dwk | 1773110 |    2793 | 12596700 |      360
 stake1ux52497mc54pczpa5vewfltxvq4r08kckcea8dndjauxaesk34u48 | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  192769 |     221 |  1779562 |      360
 stake1uy0yhskzf9cmsyyuwu04fql0dnljc8el3k3vp4jvegx7xkq88syvw | pool1kprt6l4quz03dhqx8jw22spdjvp5dkshdnn567vlexqkj35hd8a | 1044924 |    2897 |        0 |      360
 stake1u9l7es8qafpkw5ezkjc46uunhv22ma57cs24z73nxpujsxqgzgdll | pool1ek2m5zuydqcgqv3s5mlh55tr8gp76yuxe4hj4s7xtxajyc94rx3 |  203876 |     221 |        0 |      360
 stake1uyrc2hvzj6g9wkpc29fgfnd66us433ekavvegw8lq4kj5xgehhm8j | pool1kprt6l4quz03dhqx8jw22spdjvp5dkshdnn567vlexqkj35hd8a | 1189739 |    2897 |        0 |      360
(26 rows)
```
The sum of these corresponds exactly to the difference we see in the `SUM()` value:

```
select sum(amount) from grest.stake_snapshot_cache ssc where epoch_no = 360 and ssc.addr_id NOT IN (SELECT addr_id from epoch_stake where epoch_no = 362);
   sum
----------
 52807019
(1 row)
```

**Expected behaviour**

Values should match.

**Comment by dostrelith678** [comment]
These all seem to be the case of:

1. pool retires
2. pool re-registers
3. stake account does not re-delegate after pool re-registration

Hence, these stake accounts should not be captured in the snapshot.

**Comment by dostrelith678** [comment]
This is actually related to the last tx of the epoch fix (not applied in v1.0.6 for mainnet): https://github.com/cardano-community/guild-operators/pull/1490

Confirmed the snapshot is correct once the epoch last tx in cache is fixed:
```
cexplorer=# SELECT SUM(amount) FROM EPOCH_STAKE WHERE EPOCH_NO = 362;
        sum
-------------------
 24724295922908154
(1 row)

cexplorer=# SELECT SUM(amount) FROM grest.STAKE_SNAPSHOT_CACHE WHERE EPOCH_NO = 360;
        sum
-------------------
 24724295922908154
(1 row)

```
